perfmetrics/scripts/gsheet/gsheet_test.py

```diff
@@ -16,6 +17,147 @@ class MockCredentials:
+  update_response_write = {
+      'spreadsheetId': SPREADSHEET_ID,
+      'updatedRange': '{0}!A2'.format(WORKSHEET_NAME),
+      'updatedRows': 1,
+      'updatedColumns': 10,
+      'updatedCells': 10
+  }
+  sheets_service_mock = mock.MagicMock()
+  sheets_service_mock.spreadsheets().values().get(
+  ).execute.return_value = get_response
+  sheets_service_mock.spreadsheets().values().clear(
+  ).execute.return_value = update_response_clear
+  sheets_service_mock.spreadsheets().values().update(
+  ).execute.return_value = update_response_write
+  metrics_to_be_added = [
+      ('read', 50000, 40, 1653027155, 1653027215, 1, 2, 3, 4, 5)
+  ]
+  calls = [
+      mock.call.spreadsheets().values().get(
+          spreadsheetId=SPREADSHEET_ID,
+          range='{}!A1:A'.format(WORKSHEET_NAME)),
+      mock.call.spreadsheets().values().clear(
+          spreadsheetId=SPREADSHEET_ID,
+          range='{}!A2:{}'.format(WORKSHEET_NAME, last_row),
+          body={}),
+      mock.call.spreadsheets().values().update(
+          spreadsheetId=SPREADSHEET_ID,
+          valueInputOption='USER_ENTERED',
+          body={
+              'majorDimension': 'ROWS',
+              'values': [(
+                  'read', 50000, 40, 1653027155, 1653027215, 1, 2, 3, 4, 5
+              )]
+          },
+          range='{}!A2'.format(WORKSHEET_NAME))
+  ]
+
+  with mock.patch.object(gsheet, '_get_sheets_service_client'
+                         ) as get_sheets_service_client_mock:
+    get_sheets_service_client_mock.return_value = sheets_service_mock
+    gsheet.write_to_google_sheet(WORKSHEET_NAME, metrics_to_be_added)
+
+  return sheets_service_mock, calls
+
+def check_write_to_google_sheet_when_KOKORO_JOB_TYPE_env_variable_is_set(self,KOKORO_JOB_TYPE):
+  os.environ["KOKORO_JOB_TYPE"] = KOKORO_JOB_TYPE
+  sheets_service_mock,calls = call_write_to_google_sheet_function(self)
+
+  sheets_service_mock.assert_has_calls(calls, any_order=True)
+  del os.environ["KOKORO_JOB_TYPE"]
+
+def check_write_to_google_sheet_when_KOKORO_JOB_TYPE_env_variable_is_not_set(self):
+  # The logic for this function is the same as the assert_has_calls function from the python moc library.
+  sheets_service_mock,calls = call_write_to_google_sheet_function(self)
+
+  # Expected mock calls
+  expected = list(mock._CallList(sheets_service_mock._call_matcher(c) for c in calls))
+  # Actual mock calls got after calling write_to_google_sheet function
+  all_calls = list(mock._CallList(sheets_service_mock._call_matcher(c) for c in sheets_service_mock.mock_calls))
+
+  # not found stores difference between actual and expected.
+  not_found = []
+  for call in expected:
+    try:
+      all_calls.remove(call)
+    except ValueError:
+      not_found.append(call)
+
+  # not found should be equal to calls for not updating the gsheet.
+  if not_found != calls :
+    raise AssertionError("sheet is updated")
+
+def call_write_to_google_sheet_function_with_missing_permissions_raises_http_error(self):
```

Comment:
For reviewer: This is the existing logic from [master](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/perfmetrics/scripts/gsheet/gsheet_test.py#L88).

returned sheets_service_mock, metrics_to_be_added and calls to check if httpError is raised or not in both the cases(env var is set or not set).